### PR TITLE
check number of commitments in sign()

### PR DESCRIPTION
**Pull request by conradoplg**
Closes https://github.com/ZcashFoundation/frost/issues/455

Not fully sure if it's worth checking this (it's not required, the signature generation would fail anyway), but it works as a second layer of validation and it seems useful for users to be able to know what is the min_signers value by getting it from the KeyPackage. Let me know what you think.

This also adds a test for trying to sign with less than min_signers share (even bypassing this new min_signers value check) which is useful in itself and makes sure we're not interpreting thresholds wrong.

### CODE REVIEW

**Comment by conradoplg** [comment]
Solved conflicts

File: frost-core/src/tests/ciphersuite_generic.rs [hunk 1]
```diff
@@ -117,8 +119,24 @@ pub fn check_sign_with_dealer<C: Ciphersuite, R: RngCore + CryptoRng>(
         let key_package = frost::keys::KeyPackage::try_from(v).unwrap();
         key_packages.insert(k, key_package);
     }
+    // Check if it fails with not enough signers. To bypass the min_signers value
```
Comment (natalieesk) [inline frost-core/src/tests/ciphersuite_generic.rs:122]: I find this comment a bit confusing

File: frost-core/src/tests/ciphersuite_generic.rs [hunk 2]
```diff
@@ -117,8 +119,24 @@ pub fn check_sign_with_dealer<C: Ciphersuite, R: RngCore + CryptoRng>(
...
+    // check, also reduce the min_signers in KeyPackages.
```
Comment (natalieesk) [inline frost-core/src/tests/ciphersuite_generic.rs:123]: I find this comment a bit confusing

**Review by natalieesk — CHANGES_REQUESTED** [review]
Looks good. Just one thing about making a comment clearer.

File: frost-core/src/tests/ciphersuite_generic.rs (hunk 2 above)
Comment (conradoplg) [reply frost-core/src/tests/ciphersuite_generic.rs]: I expanded it in fa7f652

**Review by natalieesk — APPROVED** [review]

**Review by natalieesk — DISMISSED** [review]

**Pull request merged**